kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala

```diff
@@ -97,7 +98,8 @@ class KyuubiSessionManager private (name: String) extends SessionManager(name) {
       this,
       userConf,
       userConf.get(ENGINE_DO_AS_ENABLED),
-      parser)
+      parser,
+      tempFileService = kyuubiServer.tempFileService)
```

Comment:
do we need each session to hold one TempFileService instance? if not, just access the global TempFileService through SessionManager

Reply:
No. And all sessions share one TempFileService instance.
Here the server's `KyuubiSessionManager` passes the one "global" TempFileService in KyuubiServer to init `KyuubiSessionImpl` in server module, just like the way in Line101 for shared parser crossing sessions in `KyuubiSessionManager`.

Reply:
looks like the `parser` is an exception here. we don't need to pass a global sharable instance into each session, that makes us clear to know we are touching a sharable or session-isolated instance

for exmaple
```
class KyuubiSessionImpl {

  sessionManager.abc // we are touching a sharable instance

  conf.xxx // we are touching a session-isolated instance
}
```

Reply:
Thanks for the explanation and the example. Let me change to use the `kyuubiServer.tempFileService` global instance directly in `KyuubiSessionImpl` class.